backend/src/pages/api/admin/animal.ts

```diff
@@ -1 +1,55 @@
-export {};
+import { Types } from "mongoose";
+import { findAnimal, updateAnimal } from "server/mongodb/actions/Animal";
+import APIWrapper from "server/utils/APIWrapper";
+import { getUser } from "server/utils/Authentication";
+import { User, SubHandler, Role } from "src/utils/types";
+
+export default APIWrapper({
+  PATCH: {
+    config: {
+      requireToken: true,
+      roles: [Role.NONPROFIT_ADMIN],
+    },
+    handler: async (req) => {
+      const accessToken: string = req.headers.accesstoken as string;
+      const handler: User = await getUser(accessToken);
+
+      if (!handler) {
+        throw new Error("User not found in database!");
+      }
```

Comment:
You don't need to get the handler via the access token

the admin will have direct access to the handler id so you can just get that from the body of the request

Reply:
My bad. Was copy-pasting and forgot to take this part out